diff --git i/lsp-haskell.el w/lsp-haskell.el
index edd1ebd..7595990 100644
--- i/lsp-haskell.el
+++ w/lsp-haskell.el
@@ -188,7 +188,7 @@ These are assembled from the customizable variables
      :major-modes '(haskell-mode)
      :server-id 'hie
      ;; :multi-root t
-     :initialization-options 'lsp-haskell--make-init-options)))
+     )))
 
 (defun lsp-haskell--hie-command ()
   (funcall lsp-haskell-process-wrapper-function (lsp--haskell-hie-command)))
